File: BAEKJOON/go/1000/1400/1408.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var now, start string
	fmt.Fscanln(reader, &now)
	fmt.Fscanln(reader, &start)

	var nowSecArr, startSecArr []int

	sliceNow := strings.Split(now, ":")
	for _, num := range sliceNow {
		convertNum, _ := strconv.Atoi(num)
		nowSecArr = append(nowSecArr, convertNum)
	}
	sliceStart := strings.Split(start, ":")
	for _, num := range sliceStart {
		convertNum, _ := strconv.Atoi(num)
		startSecArr = append(startSecArr, convertNum)
	}

	nowSec := nowSecArr[0]*60*60 + nowSecArr[1]*60 + nowSecArr[2]
	startSec := startSecArr[0]*60*60 + startSecArr[1]*60 + startSecArr[2]
	var diffSec int
	if nowSec < startSec {
		diffSec = startSec - nowSec
	} else {
		diffSec = 24*60*60 - (nowSec - startSec)
	}

	intAnsArr := [3]int{int(diffSec / (60 * 60)), int((diffSec % (60 * 60)) / 60), diffSec % 60}
	var strAnsArr []string
	for _, num := range intAnsArr {
		convertStr := strconv.Itoa(num)
		if num <= 9 {
			convertStr = "0" + convertStr
		}
		strAnsArr = append(strAnsArr, convertStr)
	}
	fmt.Fprintln(writer, strings.Join(strAnsArr, ":"))
}

File: BAEKJOON/Python/1000/1400/1408.py
"""
Title : 24
Link : https://www.acmicpc.net/problem/1408
"""

from sys import stdin

input = stdin.readline


if __name__ == "__main__":
    now = list(map(int, input().strip().split(":")))
    start = list(map(int, input().strip().split(":")))

    now_sec = now[0] * 60 * 60 + now[1] * 60 + now[2]
    start_sec = start[0] * 60 * 60 + start[1] * 60 + start[2]
    if now_sec < start_sec:
        ans_sec = start_sec - now_sec
    else:
        ans_sec = 24 * 60 * 60 - (now_sec - start_sec)

    ans = [0] * 3
    ans[0] = ans_sec // (60 * 60)
    ans_sec %= 60 * 60
    ans[1] = ans_sec // 60
    ans_sec %= 60
    ans[2] = ans_sec

    make_num = lambda x: str(x).zfill(2)
    result = ":".join(map(make_num, ans))
    print(result)

'''
13:52:30
14:00:00
ans : 00:07:30

14:00:00
13:52:30
ans : 23:52:30
'''
